### PR TITLE
Alerting: Add support for configuring avatar URL for the Discord notifier

### PR DESCRIPTION
**Pull request by ChipWolf**
**What this PR does / why we need it**: Similar to the Slack alert options, this exposes an avatar URL field for Discord alerts.

### CODE REVIEW

**Comment by ChipWolf** [comment]
I failed to add a test, will add one.

**Comment by ChipWolf** [comment]
@marefr let me know if you'd prefer any other optional config to be exposed in this PR, happy to expand the scope if required.

Counterintuitively, I was also considering a proposal to remove the majority of the Discord alerting logic in favor of Slack, given [you may append](https://discord.com/developers/docs/resources/webhook#execute-slackcompatible-webhook) `/slack` to a Discord webhook. What are your thoughts?

**Comment by marefr** [comment]
@ChipWolf thanks. We should be able to accept this. Just have to check a few things internally before coming back to reviewing this one. Until then there's some missing documentation for the Discord notifier if you're interested:
- https://github.com/grafana/grafana/blob/main/docs/sources/alerting/notifications.md
- #34000

**Comment by marefr** [comment]
@ChipWolf we're currently in a migration phase why there are two instances of the discord notifier. May I ask you to sync with main branch and implement your changes in this file as well? https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/channels/discord.go

Mentioned documentation would be a really good thing to have.

**Review by marefr — CHANGES_REQUESTED** [review]
See earlier comments

**Comment by ChipWolf** [comment]
Will do @marefr
Also see https://github.com/grafana/grafana/issues/33377

File: docs/sources/alerting/notifications.md [hunk 1]
```diff
@@ -220,6 +220,19 @@ In DingTalk PC Client:
 
 6. There will be a Webhook URL in the panel, looks like this: https://oapi.dingtalk.com/robot/send?access_token=xxxxxxxxx. Copy this URL to the Grafana DingTalk setting page and then click "finish".
 
+### Discord
+
+To set up Discord, you need to create a Discord channel webhook. You can follow
+[Intro to Webhooks](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks) for a guide to do that.
+
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:227]:
```suggestion
To set up Discord, you must create a Discord channel webhook. For instructions on how to create the channel, refer to 
[Intro to Webhooks](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks) f.

```

File: docs/sources/alerting/notifications.md [hunk 2]
```diff
@@ -220,6 +220,19 @@ In DingTalk PC Client:
...
+Setting | Description
+---------- | -----------
+Webhook URL | Discord webhook URL.
+Message Content | Mention a group using @ or a user using <@ID> when notifying in a channel.
+Avatar URL | Optionally provide a URL to an image to use as the avatar for the bot's message.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:232]:
```suggestion
Avatar URL | Optionally, provide a URL to an image to use as the avatar for the bot's message.
```

File: docs/sources/alerting/notifications.md [hunk 3]
```diff
@@ -220,6 +220,19 @@ In DingTalk PC Client:
...
+
+Alternatively, you may use the [Slack](#slack) notifier by appending `/slack` to a Discord webhook URL.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:234]:
```suggestion
Alternately, use the [Slack](#slack) notifier by appending `/slack` to a Discord webhook URL.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added some copy edit suggestions.

**Comment by ChipWolf** [comment]
Thanks for reminding me my English scholarship from my childhood aged like milk @achatterjee-grafana 😄

**Comment by achatterjee-grafana** [comment]
> Thanks for reminding me my English scholarship from my childhood aged like milk @achatterjee-grafana 😄

You were concentrating on the technical details, no worries :)

**Comment by achatterjee-grafana** [comment]
This PR needs a milestone.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

File: pkg/services/ngalert/notifier/channels/discord.go [hunk 4]
```diff
@@ -34,6 +35,8 @@ func NewDiscordNotifier(model *NotificationChannelConfig, t *template.Template)
 		return nil, alerting.ValidationError{Reason: "No Settings Supplied"}
 	}
 
+	avatarURL := model.Settings.Get("avatar_url").MustString()
```
Comment (codesome) [inline pkg/services/ngalert/notifier/channels/discord.go:38]: For the ngalert, you will need to add this option [here](https://github.com/grafana/grafana/blob/91657dad182127bf577b449ff9d94e5bf86e592a/pkg/services/ngalert/notifier/available_channels.go#L669-L692
) and updated [this test](https://github.com/grafana/grafana/blob/91657dad182127bf577b449ff9d94e5bf86e592a/pkg/tests/api/alerting/api_available_channel_test.go) accordingly.

**Comment by ChipWolf** [comment]
@codesome thanks, let me know if there's anything else

**Review by marefr — APPROVED** [review]
LGTM

**Review by codesome — APPROVED** [review]

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

Suggested for this to be backported to v8.0.x branch so that it will be included in v8.0. In normal cases I would have waited with this until 8.1, but given there's a lot of changes happening around alerting for v8.0 I thought v8.0.x is more appropriate.

**Pull request merged**